rita_common/src/payment_validator/mod.rs

```diff
@@ -142,14 +182,14 @@ struct Remove {
 
 /// Removes a transaction from the pending validation queue, it may either
 /// have been discovered to be invalid or have been successfully accepted
-fn remove(msg: Remove, history: &mut PaymentValidator) {
+fn remove(msg: Remove, history: &mut PaymentValidator, sent_from_us: Option<bool>) {
     let was_present = history.unvalidated_transactions.remove(&msg.tx);
     // store successful transactions so that they can't be played back to us, at least
     // during this session
     if msg.success && was_present {
-        history
-            .successful_transactions
-            .insert(msg.tx.payment.clone().txid.unwrap());
+        let sent_from_us =
+            sent_from_us.expect("For a valid transaction, this needs to be some value!");
```

Comment:
I don't understand why this is needed? Why does it matter if we sent it or not?

Reply:
I moved the datastore from Hashset<all tx> to HashMap<id -> Hashset<tx>> (one for receiving and sent) since it was much easier to query for a particular client

For example, if a client sends me a list of txs that it made, if i stored all success transactions, I would have to iterate through all txs i have stored and get the ones that are 1.) addrssed to me 2.) sent from client. 

WIth a large exit, i didnt think it would make sense to do that, instead of just using separate hashmaps, and making small changes to how payment validator stored data

Reply:
If we just query the hashset for the txid it's O(1). So I think we can use the Hashset and have an easier time processing the data.

Reply:
Fixed